### PR TITLE
Alpha/Beta Definitions

### PR DESCRIPTION
**Pull request by doctrino**
## Description
The first call you do to a alpha/beta feature you will get the following printed in the terminal:
```
FeaturePreviewWarning: Workflow Orchestration is in alpha and is subject to breaking changes without notice. 
API version=beta, SDK version=alpha. 
See https://cognite-sdk-python.readthedocs-hosted.com/en/latest/appendix.html for more information.
```

In addition, updated docs appendix with definition of alpha and beta features.

## Checklist:
- [ ] Tests added/updated.
- [x] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [x] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [x] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite/client/utils/_experimental.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+from __future__ import annotations
+
+import warnings
+from typing import Literal
+
+
+class FeaturePreviewWarning(FutureWarning):
+    def __init__(self, api_version: Literal["alpha", "beta"], sdk_version: Literal["alpha", "beta"], feature_name: str):
+        self.api_version = api_version
```
Comment (erlendvollset) [inline cognite/client/utils/_experimental.py:9]: Maybe api_maturity and sdk_maturity instead?

File: docs/source/appendix.rst [hunk 2]
```diff
@@ -20,3 +20,21 @@ notes apply:
     fields that are not specified will not be changed. If you set 'replace', all the fields that are not
     specified, i.e., set to None and support being set to null, will be nulled out. See the API
     documentation for the update endpoint for more information.
+
+.. _appendix-alpha-beta-features:
+
+Alpha and Beta Features
```
Comment (erlendvollset) [inline docs/source/appendix.rst:26]: Excellent!

**Review by erlendvollset — APPROVED** [review]
LGTM! Can consider using "maturity" instead of "version", since version will typically mean the semantic/calendar version

File: cognite/client/utils/_experimental.py (hunk 1 above)
Comment (doctrino) [reply cognite/client/utils/_experimental.py]: Agree, that is a better word.

**Review by gregertw — APPROVED** [review]
I love this! Very good!

File: docs/source/appendix.rst [hunk 3]
```diff
@@ -20,3 +20,21 @@ notes apply:
...
+^^^^^^^^^^^^^^^^^^^^^^^^
+New Cognite Data Fusion API features may get support in the Python SDK before they are released for
+general availability (GA). These features are marked as alpha or beta in the documentation, and will also
+invoke a `FeaturePreviewWarning` when used.
+
+Furthermore, we distinguish between maturity of the API specification and the SDK implementation. Typically,
+the API specification may be in beta, while the SDK implementation is in alpha.
+
+* `aplha` - The feature is not yet released for general availability. There may be breaking changes to the API
+  specification and/or the SDK implementation without further notice.
```
Comment (haakonvt) [inline docs/source/appendix.rst:36]: a p l h a 😉

**Pull request merged**